graylog2-web-interface/src/components/event-definitions/event-definition-form/field-value-providers/TemplateFieldValueProviderForm.jsx

```diff
@@ -19,29 +15,49 @@ class TemplateFieldValueProviderForm extends React.Component {
 
   static type = 'template-v1';
 
-  handleValueChange = (event) => {
+  handleChange = (event) => {
     const { config, onChange } = this.props;
-    const nextTemplate = event.target.value;
+    const { name } = event.target;
+    const value = FormsUtils.getValueFromInput(event.target);
     const nextProviders = lodash.cloneDeep(config.providers);
     const templateProvider = nextProviders.find(provider => provider.type === TemplateFieldValueProviderForm.type);
-    templateProvider.template = nextTemplate;
-    onChange(Object.assign({}, config, { data_type: 'string', providers: nextProviders }));
+    templateProvider[name] = value;
+    onChange(Object.assign({}, config, { providers: nextProviders }));
   };
 
   render() {
     const { config } = this.props;
     const provider = config.providers.find(p => p.type === TemplateFieldValueProviderForm.type);
 
+    const helpText = (
+      <span>
+        Type a literal text to set to this Field or use{' '}
+        <ExternalLink href="https://cdn.rawgit.com/DJCordhose/jmte/master/doc/index.html">
+          JMTE syntax
+        </ExternalLink>
+        {' '}to add a dynamic Value.
```

Comment:
What do you think about adding an example here as well? Otherwise uses don't really know that they have to use the `source.` prefix to access fields from the source message.

Reply:
I didn't add an example because I was going to add that into the preview space. Also, does the template always get the results in the `source` field? I think at some point that was different for aggregations and filters, wasn't it?

Reply:
No, I think it was always `source` regardless of the event definition. I was using `source` because it references the source message/event that led to the current event that is created. Theoretically we could remove the `source.` prefix requirement. But that would be an issue should we add more information to the context at some point. So I think we should keep the prefix.

Reply:
This is the PR to remove the `key_context` from the lookup table field value provider configuration.

https://github.com/Graylog2/graylog-plugin-events/pull/64

Reply:
So, if `source` references to the source message/event that lead to the current event, what would be `source` in case of an aggregation? I think in that case there is no single event that lead to the event being generated, but a group of messages.

Reply:
@edmundoa 

For the following aggregation:

```json
{
  "group_by" : ["username", "remote_ip"],
  "series" : [
    {
      "id" : "ec167af6-3fce-4b05-b197-5422c9e1e4ef",
      "function" : "avg",
      "field" : "bytes"
    },
    {
      "id" : "ec167af6-3fce-4b05-b197-5422c9e1e4ff",
      "function" : "count",
      "field" : null
    }
  ]
}
```

you will get the following source message:

```json
{
  "username": "bernd",
  "remote_ip": "127.0.0.1",
  "aggregation_key": "bernd|127.0.0.1",
  "aggregation_value_avg_bytes": 256,
  "aggregation_value_count": 1234
}
```

We include the group field values, the key value and for every series a `aggregation_value_<function>_<field>` value.

Reply:
Ok, thank you for the details! I think this needs some screen space to be documented, and I think an inline example isn't enough in this case.

Reply:
Agreed. I think at some point it would be nice to have an endpoint that returns a template or test result for a aggregation so we can pre-fill the field spec for the user.

Reply:
Or show it as "available fields".

Reply:
👍